source/postprocess/depth_average.cc

```diff
@@ -73,25 +73,126 @@ namespace aspect
       if (this->get_time() < last_output_time + output_interval)
         return std::pair<std::string,std::string>();
 
-      const unsigned int n_statistics = 8+this->n_compositional_fields();
+      //Set up the header for the requested output variables
+      unsigned int n_statistics = 0;
+      std::vector<std::string> variables;
+      if ( output_all_variables )
+        {
+          variables.push_back ("temperature");
+          for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+            variables.push_back(std::string("C_") + Utilities::int_to_string(c));
+          variables.push_back ("adiabatic_temperature");
+          variables.push_back ("velocity_magnitude");
+          variables.push_back ("sinking_velocity");
+          variables.push_back ("Vs");
+          variables.push_back ("Vp");
+          variables.push_back ("viscosity");
+          variables.push_back ("vertical_heat_flux");
+          n_statistics = variables.size();
+        }
+      else  //we have to parse the list in this order to match the output below
+        {
+          if ( std::find( output_variables.begin(), output_variables.end(), "temperature") != output_variables.end() )
```

Comment:
I would move the ```output_all_variables``` inside each if instead and remove the whole block above (that way the order is specified once.